### PR TITLE
[PATCH v4] api: stash: add statistics counters

### PR DESCRIPTION
**Pull request by psavol**
Added statistics counters and print function to stash API for debugging support.

### CODE REVIEW

File: platform/linux-generic/odp_stash.c [hunk 1]
```diff
@@ -523,3 +524,38 @@ int odp_stash_flush_cache(odp_stash_t st)
...
+		return;
+	}
+
+	ODP_PRINT("\nStash info\n");
+	ODP_PRINT("----------\n");
+	ODP_PRINT("  handle          0x%" PRIx64 "\n", odp_stash_to_u64(st));
+	ODP_PRINT("  name            %s\n", stash->name);
+	ODP_PRINT("  index           %i\n", stash->index);
+	ODP_PRINT("  obj size        %u\n", stash->obj_size);
+	ODP_PRINT("  obj count       %u\n", stash_obj_count(stash));
```
Comment (MatiasElo) [inline platform/linux-generic/odp_stash.c:560]: Could also print the ring size or mask.

File: include/odp/api/spec/stash_types.h [hunk 2]
```diff
@@ -245,7 +249,7 @@ typedef struct odp_stash_param_t {
 	 *
 	 * See stash capabilities for supported statistics counters. Use odp_stash_stats() to read
 	 * the enabled counters. For optimal performance, enable only those counters that are
-	 * actually used.
+	 * actually used. All counters are disabled by default.
```
Comment (MatiasElo) [inline include/odp/api/spec/stash_types.h:252]: API header modified in implementation commit.

**Review by MatiasElo — APPROVED** [review]

**Comment by psavol** [comment]
v3: rebased to enable GCC 11/12 builds

**Comment by ashwinyes** [comment]
Acked-by: Ashwin Sekhar T K <asekhar@marvell.com>

**Comment by MatiasElo** [comment]
Replaced by #1625.

**Pull request closed without merging**